Makefile

```diff
@@ -19,6 +20,13 @@ install-hooks: venv
 
 vendor: Puppetfile
 	r10k puppetfile install --verbose --color
+	@echo 'You may also want to run `make .resource_types`'
+
+.resource_types: vendor
+	puppet generate types --environment ${ENVIRONMENT} --force || (\
+		echo '\e[1;31mPlease run' && \
+		echo 'mkdir -p ~/.puppetlabs/etc/puppet/ && cp /etc/skel/.puppetlabs/etc/puppet/puppet.conf ~/.puppetlabs/etc/puppet/\e[0m' && \
```

Comment:
Why not just fix this for people automatically instead of asking them to run this? Then it could also be included as part of the `vendor` target too, right?

Reply:
I don't really want to bludgeon config files in their home directory whenever `puppet generate types` fails (which could happen for some other reason). I only really put this message here for old staff that run into this error unexpectedly.

Moving it out of `vendor` is unrelated: Jenkins does `make vendor` and that will fail because it is not run on the puppetserver.

Reply:
Ah I see, that makes sense, I hadn't considered how jenkins runs this.

I think changing puppet-specific config files even in a home dir would be fine, but it's a not a big deal, it only has to be run at most once by each staff member anyway, so whatever works.